src/conduits/RwaOutputConduit3.t.sol

```diff
@@ -0,0 +1,533 @@
+
+        address quitToAddress = vm.addr(1);
+        vm.expectEmit(true, true, false, false);
+        emit File(bytes32("quitTo"), quitToAddress);
+
+        outputConduit.file(bytes32("quitTo"), quitToAddress);
+
+        assertEq(outputConduit.quitTo(), quitToAddress);
+    }
+
+    function testRevertOnFileUnrecognisedParam() public {
+        vm.expectRevert("RwaOutputConduit3/unrecognised-param");
+        outputConduit.file(bytes32("random"), address(0));
+    }
+
+    function testRevertOnUnauthorizedMethods() public {
+        vm.startPrank(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.rely(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.deny(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.hope(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.nope(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.hate(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.mate(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.kiss(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.diss(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.file(bytes32("quitTo"), address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-authorized");
+        outputConduit.yank(address(0), me, 0);
+    }
+
+    function testRevertOnNotMateMethods() public {
+        vm.startPrank(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-mate");
+        outputConduit.push();
+
+        vm.expectRevert("RwaOutputConduit3/not-mate");
+        outputConduit.quit();
+    }
+
+    function testRevertOnNotOperatorMethods() public {
+        vm.startPrank(address(0));
+
+        vm.expectRevert("RwaOutputConduit3/not-operator");
+        outputConduit.pick(address(0));
+    }
+
+    function testRevertOnPickAddressNotWhitelisted() public {
+        vm.expectRevert("RwaOutputConduit3/not-bud");
+        outputConduit.pick(vm.addr(1));
+    }
+
+    function testPush() public {
+        assertEq(outputConduit.to(), me);
+        assertEq(usdx.balanceOf(me), 0);
+        assertEq(usdx.balanceOf(address(outputConduit)), 0);
+        assertEq(dai.balanceOf(address(me)), 1000 ether);
+
+        dai.transfer(address(outputConduit), 500 ether);
+
+        assertEq(dai.balanceOf(me), 500 ether);
+        assertEq(dai.balanceOf(address(outputConduit)), 500 ether);
+
+        vm.expectEmit(true, true, false, false);
+        emit Push(address(me), 500 * USDX_BASE_UNIT);
+        outputConduit.push();
+
+        assertEq(usdx.balanceOf(address(me)), outputConduit.expectedGemAmt(500 ether));
```

Comment:
I think since `expectedGemAmt` is actually used in `push` then we better assert against a hard coded value in these tests.

We probably also want to have dedicated test/s for `expectedGemAmt` as a public function.

Reply:
Done

Reply:
Please do the same for the input conduit tests, where `gemToDai` is used.
I think hard coded expected values is better as `gemToDai` just replicates the internal calculation.